clients/spark/core/src/test/scala/io/treeverse/clients/GarbageCollectorSpec.scala

```diff
@@ -0,0 +1,152 @@
+package io.treeverse.clients
+
+import org.scalatest._
+import matchers.should._
+import funspec._
+
+import org.apache.spark.{HashPartitioner, SparkConf}
+import org.apache.spark.sql.{Dataset, SparkSession}
+
+class ARangeGetter(
+  val repo: String,
+  val commitRanges: Map[String, Seq[String]],
+  val ranges: Map[String, Seq[String]])
+    extends RangeGetter with Serializable {
+  private def verifyRepo(repo: String): Unit = {
+    if (repo != this.repo) {
+      throw new Exception(s"Expected repo ${this.repo} but got ${repo}")
+    }
+  }
+
+  def getRangeIDs(commitID: String, repo: String): Iterator[String] = {
+    verifyRepo(repo)
+    commitRanges(commitID).iterator
+  }
+  def getRangeAddresses(rangeID: String, repo: String): Iterator[String] = {
+    verifyRepo(repo)
+    ranges(rangeID).iterator
+  }
+}
+
+class GarbageCollectorSpec extends AnyFunSpec with Matchers with SparkSessionSetup {
+  describe("Spark") {
+    it("should perform Gauss summation") {
+      withSparkSession(spark => {
```

Comment:
Why is this test relevant?

Reply:
It demonstrates how to test using Spark.  Otherwise it's really hard to understand what an easy form is.  Opened #4211 to do it soon-ish.